File: total_100/ruby/dumb.rb
#!/usr/bin/env ruby
fary = Array.new(9)
if ARGV.size < 9 then
  for i in 9.times
    fary[i] = i+1.to_f
  end
else 
  for i in 9.times
    fary[i] = ARGV[i].to_f
  end
end

operators = ["+", "-", "*", "/"]

for one in operators
  for two in operators
    for three in operators
      for four in operators
        for five in operators
          for six in operators
            for seven in operators
              for eight in operators
	        formula = \
                  "#{fary[0]} #{one} #{fary[1]} #{two} " + \
                  "#{fary[2]} #{three} #{fary[3]} #{four} " + \
                  "#{fary[4]} #{five} #{fary[5]} #{six} " + \
                  "#{fary[6]} #{seven} #{fary[7]} #{eight} " + \
                  "#{fary[8]}"
		total = eval(formula) 
                if total  == 100 then
		  puts formula + " = " + String(total)
		end  
              end
            end
          end
        end
      end
    end
  end
end

File: total_100/lua/dumb.lua
#!/usr/bin/env lua5.3
expr = {}
expr[1] = '+'
expr[2] = '-'
expr[3] = '*'
expr[4] = '/'

for one = 1,4,1 do
  for two = 1,4,1 do
    for three = 1,4,1 do
      for four = 1,4,1 do
        for five = 1,4,1 do
          for six = 1,4,1 do
            for seven = 1,4,1 do
              for eight = 1,4,1 do
                assert(loadstring('total =' ..
                      '1'..expr[one].. 
                      '2'..expr[two]..
                      '3'..expr[three]..
                      '4'..expr[four]..
                      '5'..expr[five]..
                      '6'..expr[six]..
                      '7'..expr[seven]..
                      '8'..expr[eight]..
                      '9'))()
                if total == 100 then
                  print (
                       '1 '..expr[one].. 
                      ' 2 '..expr[two]..
                      ' 3 '..expr[three]..
                      ' 4 '..expr[four]..
                      ' 5 '..expr[five]..
                      ' 6 '..expr[six]..
                      ' 7 '..expr[seven]..
                      ' 8 '..expr[eight]..
                      ' 9 '..' = '..tostring(total))
                end
              end
            end
          end
        end
      end
    end
  end
end
